### PR TITLE
Initial import of ACI MultiSite module_utils

### PR DESCRIPTION
**Pull request by dagwieers**
##### SUMMARY
This PR currently includes module_utils, doc fragments and test inventory.

This is required for the upcoming MSC modules to be added.

##### ISSUE TYPE
- New Module Pull Request

##### COMPONENT NAME
msc module_utils

##### ANSIBLE VERSION
v2.8

##### ADDITIONAL INFORMATION
We have a list of functionality we are working on.

More information at:
- https://github.com/ansible/community/wiki/Network:-ACI-Community-plan
- https://github.com/ansible/community/wiki/Network:-ACI

### CODE REVIEW

**Comment by dagwieers** [comment]
@kbreit Please review !

File: lib/ansible/module_utils/network/aci/msc.py [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+        if auth['status'] != 201:
+            self.response = auth['msg']
+            self.status = auth['status']
+            self.fail_json(msg='Authentication failed: {msg}'.format(**auth))
+
+        payload = json.loads(resp.read())
+
+        self.headers['Authorization'] = 'Bearer {token}'.format(**payload)
+
+    def request(self, path, method=None, data=None):
```
Comment (kbreit) [inline lib/ansible/module_utils/network/aci/msc.py:149]: This was a copy paste and still references Meraki

File: lib/ansible/module_utils/network/aci/msc.py [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+            self.method = method
+
+        self.url = urljoin(self.baseuri, path)
+        resp, info = fetch_url(self.module,
+                               self.url,
+                               headers=self.headers,
+                               data=json.dumps(data),
+                               method=self.method,
+                               timeout=self.params['timeout'],
+                               use_proxy=self.params['use_proxy'],
```
Comment (kbreit) [inline lib/ansible/module_utils/network/aci/msc.py:163]: I think it's likely the proxy code won't work as I haven't been able to test it well under Meraki

**Review by gundalow — APPROVED** [review]
OK once other review comments have been addressed?

File: lib/ansible/module_utils/network/aci/msc.py (hunk 1 above)
Comment (dagwieers) [reply lib/ansible/module_utils/network/aci/msc.py:149]: Good catch :-)

File: lib/ansible/module_utils/network/aci/msc.py (hunk 2 above)
Comment (dagwieers) [reply lib/ansible/module_utils/network/aci/msc.py:163]: It works, at least it did for ACI because fetch_url does this for us.

Comment (dagwieers) [reply lib/ansible/module_utils/network/aci/msc.py:163]: If your system has `http_proxy` or `https_proxy` environment variables declared, fetch_url picks them up automatically. The `use_proxy` parameter would disable this behavior (which is enabled by default).

File: lib/ansible/module_utils/network/aci/msc.py [hunk 3]
```diff
@@ -0,0 +1,299 @@
...
+            except:
+                payload = json.loads(info['body'])
+            if 'code' in payload:
+                self.fail_json(msg='MSC Error {code}: {message} [{info}]'.format(**payload), payload=data)
+            else:
+                self.fail_json(msg='MSC Error:'.format(**payload), info=info, output=output)
+
+        return {}
+
+    def query_objs(self, path, **kwargs):
```
Comment (kbreit) [inline lib/ansible/module_utils/network/aci/msc.py:195]: I see you have query_objs and get_obj. What is the difference between query and get?

Comment (dagwieers) [reply lib/ansible/module_utils/network/aci/msc.py:195]: The function `get_obj()` uses `query_obj()`, so they are quite similar.

The difference is purely whether you are querying a single object, or multiple objects. If you expect only a single object, `get_obj()` is for you, if you expect multiple objects `query_objs()` is for you.

Both accept filter arguments, but some things should always return a single obj (like if name or id is unique). The only reason we have this function call is purely to ensure that when we expect a single object, and we do get more than one, we escalate to the user.

**Pull request merged**